docs/specs/saml.md

```diff
@@ -0,0 +1,697 @@
+
+  - SAML SOAP Binding is a synchronous binding which can be used in the single logout profile to send `<LogoutRequest>` to other session participants.
+  - Many other profiles can be supported if we support SOAP binding. See [SAMLConformance](https://groups.oasis-open.org/higherlogic/ws/public/download/35393/sstc-saml-conformance-errata-2.0-wd-04-diff.pdf) Section 2.
+
+We have no plan to support other bindings not mentioned at the moment.
+
+## <a id="6"></a> Signing
+
+- Authgear as an IdP always sign all issued requests, responses and assertions.
+  - The signatures are created according to [SAMLCore](https://docs.oasis-open.org/security/saml/v2.0/saml-core-2.0-os.pdf) Section 5.
+  - Each project will have independent signing certs. New certs can be generated by the portal. To facilitate cert rotations, a maximum of two certs is allowed. However, only one single cert can be used in signing. The active signing cert is controlled by the config `saml.signing.key_id`.
+  - Signing Algorithm ([SignatureMethod](https://www.w3.org/TR/xmldsig-core1/#sec-SignatureMethod)) is configurable with config `saml.signing.signature_method`. Please see below for supported values.
+  - Digest Method ([DigestMethod](https://www.w3.org/TR/xmldsig-core1/#sec-DigestMethod)) is configurable with config `saml.signing.digest_method`. Please see below for supported values.
+- If Service Providers has signing certs uploaded, all requests and responses from the Service Provider will be validated against the provided certs.
+  - To facilitate cert rotations, two certs are allowed to be uploaded. When there are two certs, the requests and responses will be considered valid if its signature can be validated by any one of the uploaded certs.
+
+### <a id="6_1"></a> Configs
+
+```yaml
+saml:
+  signing:
+    key_id: key01
+    signature_method: RSAwithSHA256
+    digest_method: SHA256
+```
+
+- `saml.signing`: Required. Configurations of signing behvaiors. MUST be an object with the following fields:
+  - `key_id`: Required. A string which points to a signing key secret inside the `saml.idp.signing` secret. See the below [Secrets] section for details.
+  - `signature_method`: Optional. An enum which specifies the signing algorithm used to generate the signature. The only supported value is `RSAwithSHA256`. If not provided, it defaults to `RSAwithSHA256`. For details, see [xmldsig](https://www.w3.org/TR/xmldsig-core1/#sec-SignatureAlg).
+  - `digest_method`: Optional. An enum which specifies the digest method used. The supported values are `SHA1` and `SHA256`. If not provided, it defaults to `SHA256`. For details, see [xmldsig](https://www.w3.org/TR/xmldsig-core1/#sec-MessageDigests).
+
+### <a id="6_2"></a> Secrets
+
+Two secrets are defined for SAML signing:
+
+```yaml
+- data:
+    certs:
+      - cert:
+          pem: "-----BEGIN CERTIFICATE-----\nMIIH/TCCBeWgAwIBAgIQaBYE3/..."
+        key:
+          alg: RS256
+          created_at: 1671439710
+          kid: key01
```

Comment:
This time, we can use the JWK thumbprint as the kid. (This is an implementation details by the way)